### PR TITLE
Bump cardano-node to 1.30.0

### PR DESCRIPTION
**Pull request by Anviking**
- [x] update stack.yaml, nix, cabal.project for 1.30.0 node release
- [x] Update readme to mention `1.30.0` tag

### Comments

Done in steps using the following commands in the cardano-node repo:
```
git diff 1.29.0 1.30.0-rc3 -- cabal.project
git diff 1.30.0-rc3 1.30.0-rc4 -- cabal.project
git diff 1.30.0-rc4 1.30.0 -- cabal.project
```

### Issue Number

ADP-1133

### CODE REVIEW

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

File: lib/shelley/src/Cardano/Wallet/Shelley/Network.hs [hunk 1]
```diff
@@ -881,11 +884,12 @@ fetchRewardAccounts tr queryRewardQ accounts = do
...
-                   fmap fromBalanceResult
-                    . LSQry
-                    . Shelley.GetFilteredDelegationsAndRewardAccounts
-                    $ Set.map toStakeCredential accounts
+        let qry = onAnyEra
+                (pure (byronValue, []))
+                shelleyQry
+                shelleyQry
+                shelleyQry
+                shelleyQry
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Network.hs:892]: With concrete eras, the constraints are much easier to handle.

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Review by piotr-iohk — APPROVED** [review]
Tests passing == lgtm

**Pull request merged**